altair/utils/core.py

```diff
@@ -471,7 +470,7 @@ def sanitize_narwhals_dataframe(
     # See https://github.com/vega/altair/issues/1027 for why this is necessary.
     local_iso_fmt_string = "%Y-%m-%dT%H:%M:%S"
     for name, dtype in schema.items():
-        if dtype == nw.Date and nw.get_native_namespace(data) is get_polars():
+        if dtype == nw.Date and nw.dependencies.is_polars_dataframe(data):
```

Comment:
`pyright` isn't happy with the typeguard here:

```
Type "polars.dataframe.frame.DataFrame | narwhals.stable.v1.DataFrame[TIntoDataFrame@sanitize_narwhals_dataframe]" is not assignable to return type "DataFrame[TIntoDataFrame@sanitize_narwhals_dataframe]"
  Type "polars.dataframe.frame.DataFrame | narwhals.stable.v1.DataFrame[TIntoDataFrame@sanitize_narwhals_dataframe]" is not assignable to type "DataFrame[TIntoDataFrame@sanitize_narwhals_dataframe]"
    "DataFrame" is not assignable to "DataFrame[TIntoDataFrame@sanitize_narwhals_dataframe]"PylancereportReturnType
```

Reporting on https://github.com/MarcoGorelli/altair/blob/308cb7a1d8c29a38d05239609b1b76d95bf6f614/altair/utils/core.py#L493

```py
return data.select(columns)
```

Reply:
that's so cool! pyright helped prevent a bug, because I should've passed `data.to_native()` here

now, to figure out why the tests are still passing, maybe test coverage needs increasing - taking a look 👀

Reply:
Looks like this wasn't explicitly being tested for - I've fixed up, and have added a test

Wow

I'll keep this example handy for the next time someone tells me that Python typing is useless and doesn't help prevent bugs 😄

Reply:
and thank you so much for spotting this

this sends chills down my spine 🥶